### DISCUSSION TITLE
Link Shared Environment Variables via REST API

### DISCUSSION THREAD

**Discussion by kbosterreasonone** [vercel/community#1522] [Idea]
### Summary

I am creating projects using the REST API. I would like to be able to use Shared Environment Variables with these projects without have to use the Dashboard to link the project and variables. However, there does not appear to be any way to link shared environment variables to a project using the API. In addition, shared environment variables also do not appear as part of the objects returned when getting a project, using `/v9/projects/{idOrName}`, or the project's environment variables specifically, `using /v9/projects/{idOrName}/env`. 

Is there a way to create this link using the API? 

Is there a way to see that a project has linked environment variables in the Project response body? or envs array response?

**Comment by amyegan** [comment]
The REST API does not currently have this ability, but I moved your post to the [Ideas](https://github.com/vercel/community/discussions/categories/ideas) category and shared it with the team so they can consider it for future development 🙇

**Reply by qunodmitrek** [reply]
Why would Vercel not provide an API for something that is available and encouraged in the UI?

**Reply by brianespinosa** [reply]
Probably because it does not even work correctly in the UI.
